cms/djangoapps/contentstore/views/course.py

```diff
@@ -1185,15 +1187,23 @@ def validate(self):
         if len(self.configuration.get('groups', [])) < 1:
             raise GroupConfigurationsValidationError(_("must have at least one group"))
 
-    def generate_id(self, used_ids):
+    @staticmethod
+    def generate_id(used_ids):
```

Comment:
I like refactoring this to be static but it is now nothing to do with group configurations. I'd move it somewhere more general purpose. Maybe it should be called ```generate_unique_sql_id``` since it uses a random number that is tuned for SQL identifiers.